ts/main.ts

```diff
@@ -0,0 +1,185 @@
+interface Equipment {
+  id: number;
+  name: string;
+}
+
+interface Muscle {
+  name: string;
+  nameEn?: string;
+  id?: number;
+}
+
+interface Exercise {
+  name: string;
+  image: string;
+  baseId: number;
+  id: number;
+  description: string;
+  primaryMuscles: Muscle[];
+  secondaryMuscles: Muscle[];
+  equipment?: Equipment;
+}
+
+const rootURL = 'https://wger.de/api/v2';
+console.log(rootURL);
+
+const $searchForm = document.querySelector('#search-form') as HTMLFormElement;
+const $views = document.querySelectorAll('section');
+const $beginBtn = document.querySelector('#begin');
+const $cardList = document.querySelector('.card-list') as HTMLDivElement;
+const $header = document.querySelector('header');
+const $hamburger = document.querySelector('.hamburger');
+const $hamburgerLinks = document.querySelector('.hamburger-links');
+const $noResults = document.querySelector('.no-results');
+
+if (!$searchForm) throw new Error('no search form found');
+if (!$views) throw new Error('no views found');
+if (!$beginBtn) throw new Error('no begin button found');
+if (!$cardList) throw new Error('no card list found');
+if (!$header) throw new Error('no header found');
+if (!$noResults) throw new Error('no results not found');
+
+function renderExercises(exerciseObj: Exercise): void {
+  const $card = document.createElement('div');
+  $card.setAttribute('class', 'card flex');
+
+  const $cardImg = document.createElement('img');
+  $cardImg.setAttribute('src', exerciseObj.image);
+  $cardImg.setAttribute('class', 'card-img');
+
+  const $cardText = document.createElement('div');
+  $cardText.setAttribute('class', 'card-text');
+  const $cardTitle = document.createElement('h3');
+  $cardTitle.textContent = exerciseObj.name;
+  const $cardCaption = document.createElement('p');
+  $cardCaption.innerHTML = exerciseObj.description;
+
+  $cardText.appendChild($cardTitle);
+  $cardText.appendChild($cardCaption);
+  $card.appendChild($cardImg);
+  $card.appendChild($cardText);
+  $cardList?.appendChild($card);
+}
```

Comment:
We should move this final append to outside of this `renderExercises` function to where your loop is calling `renderExercises`. It will allow this function to be reusable (thinking ahead about a favorites page feature)